### DISCUSSION TITLE
Implicit and automatic implementation of Deconstruct function in records of C #9.0

### DISCUSSION THREAD

**Discussion by tarikguney** [dotnet/roslyn#44931] [General] [answered]
Implicit and automatic implementation of `Deconstruct` method in classes or at least in C# 9.0 records would help to increase the adoption of Deconstruct semantics among the language users by making it more easily accessible without the extra code. All of the public properties of a data class can automatically be converted to deconstructables. Any thoughts on this?

**Comment by CyrusNajmabadi** [comment, marked as answer]
Language requests/suggustions should happen over at dotnet/csharplang.  That said, i believe ew intend to have Deconstruct automatically for positional records, but not for nominal once.
